### PR TITLE
Add MTMG support for Similarity and test for all-pairs Jaccard

### PR DESCRIPTION
**Pull request by ChuckHastings**
Added to the MTMG library support for results that contain a vertex pair and a result.  This will allow for support for all of the similarity algorithms.

Added an MTMG test that demonstrates how to call all-pairs Jaccard.

Also cleaned up the cmake files to better find ucx/ucs/ucp.

Closes https://github.com/rapidsai/graph_dl/issues/331

### CODE REVIEW

File: cpp/CMakeLists.txt [hunk 1]
```diff
@@ -77,6 +77,16 @@ rapids_find_package(CUDAToolkit REQUIRED
     INSTALL_EXPORT_SET  cugraph-exports
     )
 
+if (BUILD_CUGRAPH_MTMG_TESTS)
+    if(NOT TARGET ucx::ucp)
+        find_package(ucx REQUIRED)
```
Comment (KyleFromNVIDIA) [inline cpp/CMakeLists.txt:82]: Does UCP have its own CMake buildsystem now?

Comment (ChuckHastings) [reply cpp/CMakeLists.txt:82]: I have no idea.  We install ucx/ucp/ucs as part of the conda environment, this should just find it or fail.

Comment (KyleFromNVIDIA) [reply cpp/CMakeLists.txt:82]: Thanks. I just installed the `ucxx` package from Conda and see that it provides `ucxx-config.cmake` - that's what's needed for this to work.

**Review by KyleFromNVIDIA — APPROVED** [review]

**Review by seunghwak — APPROVED** [review]
LGTM

**Pull request merged**